satellite/satellitedb/overlaycache.go

```diff
@@ -902,6 +904,67 @@ func (cache *overlaycache) GetExitStatus(ctx context.Context, nodeID storj.NodeI
 	return exitStatus, Error.Wrap(err)
 }
 
+// GetGracefulExitCompletedByTimeFrame returns nodes who have completed graceful exit within a time window (time window is around graceful exit completion).
+func (cache *overlaycache) GetGracefulExitCompletedByTimeFrame(ctx context.Context, begin, end time.Time) (exitedNodes storj.NodeIDList, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	rows, err := cache.db.Query(cache.db.Rebind(`
+		SELECT id FROM nodes
+		WHERE exit_initiated_at IS NOT NULL
+		AND exit_finished_at IS NOT NULL
+		AND exit_finished_at >= ?
+		AND exit_finished_at < ?
+		`), begin, end,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = errs.Combine(err, rows.Close())
+	}()
+
+	for rows.Next() {
+		var id storj.NodeID
+		err = rows.Scan(&id)
+		if err != nil {
+			return nil, err
+		}
+		exitedNodes = append(exitedNodes, id)
+	}
+	return exitedNodes, nil
+}
+
+// GetGracefulExitIncompleteByTimeFrame returns nodes who have initiated, but not completed graceful exit within a time window (time window is around graceful exit initiation).
+func (cache *overlaycache) GetGracefulExitIncompleteByTimeFrame(ctx context.Context, begin, end time.Time) (exitingNodes storj.NodeIDList, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	rows, err := cache.db.Query(cache.db.Rebind(`
+		SELECT id FROM nodes
+		WHERE exit_initiated_at IS NOT NULL
+		AND exit_finished_at IS NULL
+		AND exit_initiated_at >= ?
+		AND exit_initiated_at < ?
+		`), begin, end,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = errs.Combine(err, rows.Close())
+	}()
+
+	// TODO return more than just ID
```

Comment:
If this TODO is something we care about, you could get it by making this a dbx query instead, and return everything the same way `Get()` does

Reply:
I'm not sure how important implementing this TODO is - if the number of nodes graceful exiting at a time is pretty small, it shouldn't be too bad to do an additional `overlaycache.Get()` for each gracefully exiting node. Otherwise, we might need to do something like you described.